src/components/flyout/flyout.spec.tsx

```diff
@@ -96,4 +96,70 @@ describe('EuiFlyout', () => {
         });
     });
   });
+
+  describe('focusTrapProps', () => {
+    const FlyoutComplex = ({
+      children = childrenDefault,
+      useShards = true,
+      focusTrapProps = {},
+    }) => {
+      const outsideRef = useRef();
+      const [isOpen, setIsOpen] = useState(false);
+
+      return (
+        <>
+          {isOpen ? (
+            <EuiFlyout
+              data-test-subj="flyoutSpec"
+              onClose={() => setIsOpen(false)}
+              ownFocus={false}
+              outsideClickCloses={true}
+              focusTrapProps={{
+                shards: useShards ? [outsideRef] : [],
+                ...focusTrapProps,
+              }}
+            >
+              {children}
+            </EuiFlyout>
+          ) : null}
+          <button
+            data-test-subj="toggle"
+            onClick={() => setIsOpen((isOpen) => !isOpen)}
+          >
+            Toggle
+          </button>
+          <button ref={outsideRef} data-test-subj="outside">
+            Outside
+          </button>
+        </>
+      );
+    };
+    it('does not close the flyout when a shard is clicked', () => {
+      cy.mount(<FlyoutComplex />);
+      cy.get('[data-test-subj="toggle"]')
+        .realClick()
+        .then(() => {
+          expect(cy.get('[data-test-subj="flyoutSpec"]').should('exist'));
+        });
+      cy.get('[data-test-subj="outside"]')
+        .realClick()
+        .then(() => {
+          expect(cy.get('[data-test-subj="flyoutSpec"]').should('exist'));
+        });
+    });
+
+    it('delays the close callback when the focus trap is set to close on mouseup instead of mousedown', () => {
+      cy.mount(<FlyoutComplex focusTrapProps={{ closeOnMouseup: true }} />);
```

Comment:
Sorry, I missed the part where `useShards` is necessary b/c of the ref.
```suggestion
    it('delays the close callback when the focus trap is set to close on mouseup instead of mousedown', () => {
      cy.mount(<FlyoutComplex focusTrapProps={{ closeOnMouseup: true }} useShards={false} />);
```

Honestly I'd also be open to removing these 2 tests now that you've added `focus_trap` tests. I feel like the focus trap tests sufficiently cover this behavior especially since EuiFlyout is just a passthrough for it, and since EuiCollapsibleNav already covers the shards implementation. Thoughts?

Reply:
Removed these tests